SimPEG/MT/SurveyMT.py

```diff
@@ -0,0 +1,589 @@
+                    Tij = sDiag(aHd*( sDiag(abz_py)*abx_px - sDiag(abz_px)*abx_py))
+                    TijN_uV = lambda x: abx_px_u(sDiag(abz_py)*x) + abz_py_u(sDiag(abx_px)*x) - abx_py_u(sDiag(abz_px)*x) - abz_px_u(sDiag(abx_py)*x)
+                # Calculate the complex derivative
+                PDeriv_real = TijN_uV(aHd*v) - aHd_uV(Tij.T*aHd*v)#
+                # NOTE: Need to reshape the output to go from 2*nU array to a (nU,2) matrix for each polarization
+                # PDeriv_real = np.hstack((mkvc(PDeriv_real[:len(PDeriv_real)/2],2),mkvc(PDeriv_real[len(PDeriv_real)/2::],2)))
+                PDeriv_real = PDeriv_real.reshape((2,mesh.nE)).T
+            # Extract the data
+            if real_or_imag == 'imag':
+                Pv = 1j*PDeriv_real
+            elif real_or_imag == 'real':
+                Pv = PDeriv_real.astype(complex)
+
+
+        return Pv
+
+#################
+###  Survey   ###
+#################
+class Survey(SimPEGsurvey.BaseSurvey):
+    """
+        Survey class for MT. Contains all the sources associated with the survey.
+
+        :param list srcList: List of sources associated with the survey
+
+    """
+    srcPair = SrcMT.BaseMTSrc
+
+    def __init__(self, srcList, **kwargs):
+        # Sort these by frequency
+        self.srcList = srcList
+        SimPEGsurvey.BaseSurvey.__init__(self, **kwargs)
+
+        _freqDict = {}
+        for src in srcList:
+            if src.freq not in _freqDict:
+                _freqDict[src.freq] = []
+            _freqDict[src.freq] += [src]
+
+        self._freqDict = _freqDict
+        self._freqs = sorted([f for f in self._freqDict])
+
+    @property
+    def freqs(self):
+        """Frequencies"""
+        return self._freqs
+
+    @property
+    def nFreq(self):
+        """Number of frequencies"""
+        return len(self._freqDict)
+
+    # TODO: Rename to getSources
+    def getSrcByFreq(self, freq):
+        """Returns the sources associated with a specific frequency."""
+        assert freq in self._freqDict, "The requested frequency is not in this survey."
+        return self._freqDict[freq]
+
+    def projectFields(self, u):
+        data = Data(self)
+        for src in self.srcList:
+            sys.stdout.flush()
+            for rx in src.rxList:
+                data[src, rx] = rx.projectFields(src, self.mesh, u)
+        return data
+
+    def projectFieldsDeriv(self, u):
+        raise Exception('Use Transmitters to project fields deriv.')
+
+#################
+###   Data    ###
+#################
+class Data(SimPEGsurvey.Data):
+    '''
+    Data class for MTdata
+
+    :param SimPEG survey object survey:
+    :param v vector with data
+
+    '''
+    def __init__(self, survey, v=None):
+        # Pass the variables to the "parent" method
+        SimPEGsurvey.Data.__init__(self, survey, v)
+
+    # # Import data
+    # @classmethod
+    # def fromEDIFiles():
+    #     pass
+
+    def toRecArray(self,returnType='RealImag'):
```

Comment:
this is cool, we will have to chat about abstracting this in the new and improved data class :)